### PR TITLE
Bug fix for setting up data, and training of emulator. 

### PR DESCRIPTION
**Pull request by taegyunkim**

### CODE REVIEW

**Comment by Atcold** [comment]
For the greek letters, like ϕ, type `\phi` and press *Tab* in the Jupyter notebook.

File: 14-truck_backer-upper.ipynb [hunk 1]
```diff
@@ -364,12 +364,12 @@
     "\n",
     "for i in tqdm(torch.randperm(len(train_inputs))):\n",
     "    ϕ_state = train_inputs[i]\n",
+    "    optimiser_e.zero_grad()\n",
```
Comment (Atcold) [inline 14-truck_backer-upper.ipynb]: Why?

File: 14-truck_backer-upper.ipynb [hunk 2]
```diff
@@ -253,7 +253,7 @@
     "\n",
     "episodes = 10\n",
     "data_set = list()\n",
-    "# truck = Truck(); episodes = 10_000  # uncooment for creating the data set\n",
+    "truck = Truck(); episodes = 10_000  # uncooment for creating the data set\n",
```
Comment (Atcold) [inline 14-truck_backer-upper.ipynb]: Plz, get this back to how it was.
Uncommenting should be done by the user.

File: 14-truck_backer-upper.ipynb (hunk 1 above)
Comment (taegyunkim) [reply 14-truck_backer-upper.ipynb]: What do you mean why?
Are you asking why did I put this here?

File: 14-truck_backer-upper.ipynb [hunk 3]
```diff
@@ -330,8 +341,8 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "train_data = (tensor_train_set - mean) / std\n",
-    "test_data = (tensor_test_set - mean) / std"
+    "train_inputs = (tensor_train_inputs - mean) / std\n",
+    "test_inputs = (tensor_test_inputs - mean) / std"
```
Comment (Atcold) [inline 14-truck_backer-upper.ipynb]: Apply normalisation also to the output tensors.

File: 14-truck_backer-upper.ipynb (hunk 1 above)
Comment (Atcold) [reply 14-truck_backer-upper.ipynb]: Yes, why did you move it up?

Comment (taegyunkim) [reply 14-truck_backer-upper.ipynb]: Probably I did it without knowing what I was doing.

Comment (Atcold) [reply 14-truck_backer-upper.ipynb]: It does not change anything. But I like to have it just before `.backward()`.

**Comment by Atcold** [comment]
When you've applied these edits, I'll merge this.
Plear, open a new PR for the controller.

**Comment by taegyunkim** [comment]
Please take a look. Resolved all of your comments.

**Comment by Atcold** [comment]
You don't need to use input and output mean and std. They are basically the same, right?
I think you can use the stats of the input for normalising everything. So that people don't get confused, later, with the training of the controller.

**Comment by taegyunkim** [comment]
> You don't need to use input and output mean and std. They are basically the same, right?
> I think you can use the stats of the input for normalising everything. So that people don't get confused, later, with the training of the controller.
> 
> What do you think?

Right, should be fine to do so. Updated.

**Comment by Atcold** [comment]
Thanks! Waiting to see a working controller now!

**Comment by taegyunkim** [comment]
> Thanks! Waiting to see a working controller now!

I realized that simply using the truck class isn't trivial as it's using non pytorch functions to compute cos, sin. Probably I need to do some modification to Truck or define another set of helper function to compute gradient.

**Comment by Atcold** [comment]
You don't need to compute any gradient through the controller.
The gradients will flow through the emulator.

**Comment by taegyunkim** [comment]
Doesn't it need to flow through the state? I was using truck class to store and pass states.

**Comment by Atcold** [comment]
In the paper they use the emulator to pass back the gradient.
They use the truck kinematics as teacher forcing.
